### PR TITLE
feat: add utility template functions (#272)

### LINKED ISSUE #272: Feature: Add sprig templating functions
Consider adding [sprig](https://github.com/Masterminds/sprig) templating functions to extend capabilities. Originally suggested upstream https://github.com/kelseyhightower/confd/issues/770

### PR DESCRIPTION
**Pull request by abtreece**
## Summary

Adds 28 utility template functions to confd, implemented with Go standard library only (zero new dependencies). Inspired by common patterns from Sprig/Helm but cherry-picked for config management use cases.

- **Defaults & conditionals**: `default`, `ternary`, `coalesce`, `empty`
- **JSON**: `toJson`, `fromJson`, `toPrettyJson`
- **Formatting**: `indent`, `nindent`, `quote`, `squote`
- **Regex**: `regexMatch`, `regexFind`, `regexReplaceAll`
- **Strings**: `trimPrefix`, `repeat`, `nospace`, `snakecase`, `camelcase`, `kebabcase`
- **Collections**: `dict`, `list`, `hasKey`, `keys`, `values`, `append`, `pluck`
- **Hashing**: `sha256sum`

### Design decisions

- **No external dependencies** — all functions use Go standard library only
- **confd functions win on conflict** — utility functions loaded as base layer, confd's existing functions overlay on top
- **Regex functions return errors** instead of panicking on invalid patterns
- **`dict`** is an alias for the existing `map`/`CreateMap` function (users expect the Sprig/Helm name)
- **`keys`** returns sorted keys for deterministic template output

Closes #272

## Test plan

- [x] Table-driven unit tests for all 28 functions with edge cases
- [x] Integration test rendering a Go template with new + existing functions together
- [x] `go test -race ./pkg/...` passes
- [x] `go vet ./...` clean
- [x] Existing tests unchanged — no regressions
- [ ] CI passes

### CODE REVIEW

File: pkg/template/utility_funcs.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+func camelcase(s string) string {
+	words := splitWords(s)
+	if len(words) == 0 {
+		return ""
+	}
+	for i, w := range words {
+		if i == 0 {
+			words[i] = strings.ToLower(w)
+		} else {
+			words[i] = strings.ToUpper(w[:1]) + strings.ToLower(w[1:])
```
[the remark this answers is not in the document]
Comment (abtreece) [reply pkg/template/utility_funcs.go]: Fixed — camelcase now uses []rune and unicode.ToUpper for the first character instead of byte slicing, making it safe for non-ASCII input.
